controllers/code.go

```diff
@@ -1,24 +1,43 @@
 package controllers
 
 import (
+	"errors"
+	"io/ioutil"
 	"net/http"
 
+	"github.com/coduno/api/util"
+	"github.com/gorilla/mux"
+
+	"google.golang.org/cloud/storage"
+
 	"golang.org/x/net/context"
 )
 
-// DownloadTemplate serves a static file to a client.
+// Template serves the contents of a static file to a client.
 // TODO(flowlo, victorbalan): Decide where the templates will be stored.
-func Template(c context.Context, w http.ResponseWriter, r *http.Request) (int, error) {
+func Template(ctx context.Context, w http.ResponseWriter, r *http.Request) (int, error) {
 	if r.Method != "GET" {
 		return http.StatusMethodNotAllowed, nil
 	}
 
-	// TODO(victorbalan): Serve correct template using passenger and result key.
+	templateName := mux.Vars(r)["name"]
+	language := mux.Vars(r)["language"]
+	fileName, ok := util.FileNames[language]
+
+	if !ok {
+		return http.StatusInternalServerError, errors.New("language unknown")
+	}
 
-	// FIXME(victorbalan): Send correct Content-Type
-	w.Header().Set("Content-Type", "application/octet-stream")
-	w.Header().Set("Content-Disposition", "attachment; filename='template.java'")
```

Comment:
Ignore the code editor. This is an API endpoint to download the template. So we should set reasonable headers. If you are too lazy to implement this now, you can leave it as-is, but it does not justify just deleting a FIXME without comment.